include/exadg/fluid_structure_interaction_precice/exadg_interface.h

```diff
@@ -48,10 +46,15 @@ class ExaDGInterface : public CouplingInterface<dim, data_dim, VectorizedArrayTy
   write_data(const LinearAlgebra::distributed::Vector<double> & data_vector,
              const std::string &                                data_name) override;
 
-  virtual std::vector<Tensor<1, dim>>
+  virtual void
   read_block_data(const std::string & data_name) const override;
 
+  void
+  set_data_pointer(std::shared_ptr<ExaDG::InterfaceCoupling<dim, dim, double>> exadg_terminal_);
```

Comment:
Is this function public? If so, why not private?

Reply:
Yes it is. For compatibility reasons I didn't want to pass the pointers via the Constructors since this would require to adjust the base class. Hence, I introduced an additional public functions in order to set the pointers.